packages/@sanity/presence-store/src/PresenceStore.js

```diff
@@ -0,0 +1,152 @@
+/* global window */
+
+import Observable from '@sanity/observable/minimal'
+import UUID from '@sanity/uuid'
+import deepEqual from 'deep-equal'
+
+const RESEND_REPORT_INTERVAL = 15000
+const STATE_TIMEOUT_INTERVAL = RESEND_REPORT_INTERVAL * 2
+
+function hashKeyForState(identity, session) {
+  return `${identity}__${session}`
+}
+
+export default class PresenceStore {
+  constructor(connection, channel) {
+    this.connection = connection
+    this.unsubscribeListener = this.connection.listen().subscribe(this.handleMessage).unsubscribe
+    this.myState = {}
+    this.states = {}
+    this.timestamps = {}
+    this.resendReportTimer = null
+    this.changeReportDebounceTimer = null
+    this.performPurgeTimer = setInterval(this.performPurge, 2000)
+    this.sessionId = UUID()
+    this.requestRollCall()
+
+    // List of functions that will be called with presence change notifications
+    this.subscriberCallbacks = []
+  }
+
+  // Close Presence Store, should be called when window closes
+  close() {
+    this.connection.sendBeacon({type: 'disconnect', session: this.sessionId})
+    this.unsubscribeListener()
+    clearTimeout(this.resendReportTimer)
+    clearInterval(this.performPurgeTimer)
+  }
```

Comment:
There are a couple more timers that should be closed here, I think? `changeReportDebounceTimer` for instance